### PR TITLE
Implement error code handling in lifecyclePolicy

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
**What this PR does / why we need it**:

Implement error code handling in lifecyclePolicy

**Special notes for your reviewer**:

**Release note**:

```release-note
Implement error code handling in lifecyclePolicy.
```

### CODE REVIEW

File: pkg/controllers/job/job_controller_handler.go [hunk 1]
```diff
@@ -212,9 +212,14 @@ func (cc *Controller) updatePod(oldObj, newObj interface{}) {
 	}
 
 	event := vkbatchv1.OutOfSyncEvent
+	var exitCode int32
 	if oldPod.Status.Phase != v1.PodFailed &&
 		newPod.Status.Phase == v1.PodFailed {
 		event = vkbatchv1.PodFailedEvent
+		// TODO: currently only one container pod is supported by volcano
+		// Once multi containers pod is supported, update accordingly.
+		exitCode = newPod.Status.ContainerStatuses[0].State.Terminated.ExitCode
```
Comment (k82cn) [inline pkg/controllers/job/job_controller_handler.go:221]: let's open an issue to trace that. one option is to set something like "1:1" which means considering second container's exist code (or by container name, e.g. worker:-2)

Comment (k82cn) [reply pkg/controllers/job/job_controller_handler.go:221]: the default value is 0, the "first" container.

**Comment by k82cn** [comment]
LGTM overall

File: pkg/controllers/job/job_controller_handler.go (hunk 1 above)
Comment (hzxuzhonghu) [reply pkg/controllers/job/job_controller_handler.go:221]: I will open an issue following this.

**Pull request merged**